contracts/YieldComponentToken.sol

```diff
@@ -154,24 +155,59 @@ contract YieldComponentToken is ERC20Base, Ownable {
   /// @dev Internal yield payout function that computes the yield and transfers it to the owner
   /// @param owner Owner and recipient of the accrued yield
   function _payoutYield(address owner) private {
-    uint256 lastPrice = lastPrices[owner];
-    uint256 balance = balances[owner];
     uint256 currPrice = priceOracle.getPrice(fullToken);
     // Make sure the last price is always updated when paying out yield.
     lastPrices[owner] = currPrice;
-    if (balance == 0 || lastPrice == 0) {
+    uint256 payoutAmount = calculatePayoutAmount(owner, currPrice);
+    if (payoutAmount == 0) {
       return;
     }
+    // Call the payout function on the SplitVault contract, just for the yield
+    splitVault.payout(payoutAmount, fullToken, owner);
+  }
+
+  /// @dev Simplest public method for calculating the outstanding yield for a yield token holder
+  /// @param owner Owner and future recipient of the accrued yield
+  /// @return The payout amount denoted in the decimal precision of the fullToken
+  function calculatePayoutAmount(address owner) public view returns (uint256) {
+    uint256 currPrice = priceOracle.getPrice(fullToken);
+    return calculatePayoutAmount(owner, currPrice);
+  }
+
+  /// @dev Public method for calculating the outstanding yield for a yield token holder and a new fullToken price
+  /// @param owner Owner and future recipient of the accrued yield
+  /// @param currPrice The price of fullToken to use for the calculation. Must be more than internally stored lastPrice
+  /// @return The payout amount denoted in the decimal precision of the fullToken
+  function calculatePayoutAmount(address owner, uint256 currPrice) public view returns (uint256) {
+    uint256 lastPrice = lastPrices[owner];
+    uint256 balance = balances[owner];
+    if (balance == 0 || lastPrice == 0) {
+      return 0;
+    }
+    uint256 payoutAmount = calculatePayoutAmount(balance, currPrice, lastPrice, ERC20(fullToken).decimals());
```

Comment:
Not all ERC20's are properly implemented and some might lack the `decimals()` method. I'm assuming in those cases this method would revert. If so, we should probably check for this method in the `split()` method and disallow the splitting of fullTokens that don't have it.

Reply:
Adding to constructor for both capital and yield.